File: bismmillah/E-learning_website-main (4)/E-learning_website-main/backend/microservice/src/main/java/tn/esprit/microservice/services/QuizQuestionService.java
package tn.esprit.microservice.services;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import tn.esprit.microservice.entities.QuizQuestion;
import tn.esprit.microservice.repositories.QuizQuestionRepository;

import java.util.List;


@Service
@RequiredArgsConstructor
public class QuizQuestionService {
    @Autowired
    private QuizQuestionRepository questionRepository;

    // Create a new question
    public QuizQuestion createQuestion(QuizQuestion question) {
        return questionRepository.save(question);
    }

    // Get all questions for a specific quiz
    public List<QuizQuestion> getQuestionsByQuizId(Long quizId) {
        return questionRepository.findByQuizId(quizId);
    }

    // Get a question by ID
    public QuizQuestion getQuestionById(Long id) {
        return questionRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Question not found"));
    }

    // Update a question
    public QuizQuestion updateQuestion(Long id, QuizQuestion questionDetails) {
        QuizQuestion question = getQuestionById(id);
        question.setText(questionDetails.getText());
        return questionRepository.save(question);
    }

    // Delete a question
    public void deleteQuestion(Long id) {
        questionRepository.deleteById(id);
    }
}

File: frontend/backoffice/src/service/mic/quiz-question.service.ts
// File: src/app/services/quiz-question.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface QuizQuestion {
  id: number;
  text: string;
  quizId: number;
}

@Injectable({
  providedIn: 'root',
})
export class QuizQuestionService {
  private baseUrl = 'http://localhost:8088/questions'; // Backend URL

  constructor(private http: HttpClient) {}

  // Get all questions for a specific quiz
  getQuestionsByQuizId(quizId: number): Observable<QuizQuestion[]> {
    return this.http.get<QuizQuestion[]>(`${this.baseUrl}/quiz/${quizId}`);
  }

  // Create a new question
  createQuestion(question: QuizQuestion): Observable<QuizQuestion> {
    return this.http.post<QuizQuestion>(`${this.baseUrl}/create`, question);
  }

  // Delete a question
  deleteQuestion(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}